lib/semian/simple_enum.rb

```diff
@@ -0,0 +1,29 @@
+module Semian
+  module Simple
+    class Enum #:nodoc:
+      def initialize(symbol_list:)
+        @allowed_symbols = symbol_list
+        reset
+      end
+
+      attr_reader :value
+
+      def value=(sym)
+        raise ArgumentError unless @allowed_symbols.include?(sym)
```

Comment:
What do you mean? It already blocks invalid values, otherwise there isn't much ...